CONTRIBUTE.md

```diff
@@ -0,0 +1,333 @@
+# JavaScript Interactive Examples Contributing Guide
+
+A guide for people wanting to contribute interactive JavaScript examples.
+
+## Naming Things
+
+It probably makes sense to name things in a consistent way throughout all examples.  Where possible the naming should ‘reflect’ the type of that object or variable.
+
+### Numerical Suffixes
+
+One way we could maintain a certain consistency is to use the type of the object with a numerical suffix.
+
+For example:
+
+```
+object1 = {};
+var string1 = “a string”;
+var array1 = ['a', 'b', 'c'];
+var array2 = [1, 2, 3];
+```
+
+Note it is suggested that even if there is only one instance of that object then it should be suffixed with a `1`.
+
+Note also although we could use the same name for multiple instances, it's suggested that we use a progressive numbering system to minimise confusion should the reader want to experiment with the example code.
+
+For example:
+
+```
+var array1 = ['a', 'b', 'c'];
+
+console.log(array1);
+// expected output: Array ["a", "b", "c"]
+
+var array2 = [1, 2, 3];
+
+console.log(array2);
+// expected output: Array [1, 2, 3]
+```
+
+An alternative to using numerical suffixes could be to use more descriptive suffixes, such as `arrayLetters` and `arrayNumbers` in the above examples.
+
+This may make more sense in the context of the following example:
+
+```
+var collatorDe = new Intl.Collator('de', { usage: 'search', sensitivity: 'base' });
+var collatorFr = new Intl.Collator('fr', { usage: 'search', sensitivity: 'base' });
+```
+
+### Indicating console.log output
+
+To indicate the ouput we expect, we place a comment `// expected output:` on the line below each `console.log`.
+
+
+### Representing Browser Differences
+
+Occasionally browsers will produce different results, in these cases we can denote the varying output like so:
+
+```
+console.log(matchesDe);
+// expected output (Chrome / Firefox): Array ["Bären"]
+// expected output (Safari): Array ["Bären", "Baren"]
+```
+
+## Language Choice
+
+For more established example content (such as arrays) it is recommended that we stick to ES5, where examples are required for APIs standardised after ES6 it is felt that we can use ES6 to illustrate the examples.
+
+ES6 examples should use:
+
+- `let` and `const` instead of `var`
+- Arrow functions (`=>`) for Anonymous Functions (see below)
+- Template literals (``string text ${expression} string text``)
+- Spread syntax (`myFunction(...iterableObj);`)
+
+## Providing Context
+
+Where possible it may be useful to provide examples showing how a method could be used in the context of a function or class. It may not always be possible to provide a succinct example and a series of `console.log`s can also be a valid way of illustrating an example.
+
+## Tangible examples
+
+It's nice to be able to illustrate a programming concept using real or imaginary world examples, if possible using the same theme throughout a specific topic.
+
+For example:
+
+```
+function 1(disposition) {
```

Comment:
Is "1" here and below intentional, or do you mean "monster1"?

Reply:
Yup - the monster part got eaten somehow.